### PR TITLE
Add client side encryption

### PR DESCRIPTION
**Pull request by interstellarPotato**

### CODE REVIEW

File: src/test/java/org/gaul/s3proxy/EncryptedBlobStoreTest.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+                .newBuilder("transient")
+                .credentials("identity", "credential")
+                .modules(ImmutableList.<Module>of(new SLF4JLoggingModule()))
+                .build(BlobStoreContext.class);
+        blobStore = context.getBlobStore();
+        blobStore.createContainerInLocation(null, containerName);
+
+        properties = new Properties();
+        properties.put(PROPERTY_ENCRYPTED_BLOBSTORE, "true");
+        properties.put(PROPERTY_PEACEMAKR_API_KEY, "");
```
Comment (notasecret) [inline src/test/java/org/gaul/s3proxy/EncryptedBlobStoreTest.java]: should just work?

Comment (notasecret) [reply src/test/java/org/gaul/s3proxy/EncryptedBlobStoreTest.java]: could org test api key

Comment (notasecret) [reply src/test/java/org/gaul/s3proxy/EncryptedBlobStoreTest.java]: ie, https://api.peacemakr.io/api/v1/org/key/test

File: src/main/java/org/gaul/s3proxy/EncryptedBlobStore.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        newBlob.getMetadata().setPublicUri(blobMeta.getPublicUri());
+        newBlob.getMetadata().setContainer(blobMeta.getContainer());
+
+        return newBlob;
+    }
+
+    private Blob encryptBlob(String container, Blob blob) {
+        try {
+            byte[] payload = getByteArrayFromStream(blob);
+            byte[] encrypted = peacemakrSDK.encryptInDomain(payload, "works2");
```
Comment (dahuang37) [inline src/main/java/org/gaul/s3proxy/EncryptedBlobStore.java]: Maybe we should use "default" domain or mention this in the README? :)

Comment (interstellarPotato) [reply src/main/java/org/gaul/s3proxy/EncryptedBlobStore.java]: Agree :)

File: src/test/java/org/gaul/s3proxy/EncryptedBlobStoreTest.java (hunk 1 above)
Comment (interstellarPotato) [reply src/test/java/org/gaul/s3proxy/EncryptedBlobStoreTest.java]: We no longer need to provide an api key to run local-test only.

**Comment by notasecret** [comment]
Screen Shot 2021-06-02 at 9 55 00 AM
Demo works great. Thx @interstellarPotato

**Pull request merged**